### PR TITLE
Disable some impactful analyzers

### PR DESCRIPTION
**Pull request by Rob-Hague**
I promise to stop whining about analyzers if this is merged.

Before:

image

After:

image

Done with a clean build each time (`git clean -xdf`). The individual numbers are quite variable, but it is around 60% build time reduction (still much longer than it used to be).

### CODE REVIEW

**Comment by WojciechNagorski** [comment]
LGTM! I think it's good compromise.
@drieseng it's your topic? Are you agree?

File: .editorconfig [hunk 1]
```diff
@@ -263,6 +351,9 @@ dotnet_diagnostic.SA1413.severity = none
 # This is a duplicate of IDE0011.
 dotnet_diagnostic.SA1503.severity = none
 
+# SA1512: Single-line comments should not be followed by a blank line
```
Comment (drieseng) [inline .editorconfig:386]: Please motivate. I'm ok with this though.

File: .editorconfig [hunk 2]
```diff
@@ -198,6 +256,30 @@ dotnet_diagnostic.S4487.severity = none
 # This is a partial duplicate of MA0067, and we do not want to report the use of 'default' for a Guid as error.
 dotnet_diagnostic.S4581.severity = none
 
+# S4663: Comments should not be empty
+# https://rules.sonarsource.com/csharp/RSPEC-4663/
+dotnet_diagnostic.S4663.severity = none
```
Comment (drieseng) [inline .editorconfig]: I prefer to keep this one enabled.

File: .editorconfig [hunk 3]
```diff
@@ -168,6 +212,10 @@ dotnet_diagnostic.S3925.severity = none
 # This is a duplicate of MA0015.
 dotnet_diagnostic.S3928.severity = none
 
+# S3949: Calculations should not overflow
+# https://rules.sonarsource.com/csharp/RSPEC-3949/
+dotnet_diagnostic.S3949.severity = none
```
Comment (drieseng) [inline .editorconfig:253]: Please motivate.

File: .editorconfig [hunk 4]
```diff
@@ -83,6 +83,16 @@ dotnet_diagnostic.S1481.severity = none
...
 
+# S1944: Invalid casts should be avoided
+# https://rules.sonarsource.com/csharp/RSPEC-1944/
+#
+# Disabled due to build performance impact.
+dotnet_diagnostic.S1944.severity = none
+
+# S2053: Hashes should include an unpredictable salt
+# https://rules.sonarsource.com/csharp/RSPEC-2053/
+dotnet_diagnostic.S2053.severity = none
```
Comment (drieseng) [inline .editorconfig:103]: Please motivate.

File: .editorconfig [hunk 5]
```diff
@@ -104,8 +114,8 @@ dotnet_diagnostic.S2551.severity = none
 # S2583: Conditionally executed code should be reachable
 # https://rules.sonarsource.com/csharp/RSPEC-2583
 #
-# This rule produces false errors in, for example, for loops.
```
Comment (drieseng) [inline .editorconfig:119]: Please also keep the original motivation.

**Comment by drieseng** [comment]
> The individual numbers are quite variable, but it is around 60% build time reduction (still much longer than it used to be).

Is that also the explanation for the Meziantou numbers? As far as I can tell, you didn't disable any of it rules.

**Review by drieseng — CHANGES_REQUESTED** [review]

**Comment by Rob-Hague** [comment]
> Is that also the explanation for the Meziantou numbers? As far as I can tell, you didn't disable any of it rules.

Yes I believe so, I don't have a better explanation

(sorry, somehow I edited your comment to this comment...)

**Review by drieseng — APPROVED** [review]

**Comment by WojciechNagorski** [comment]
The 2023.0.1 version has been released to Nuget: https://www.nuget.org/packages/SSH.NET/2023.0.1

**Pull request merged**